core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/FiltersAggregator.java

```diff
@@ -86,11 +91,16 @@ public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
         return new LeafBucketCollectorBase(sub, null) {
             @Override
             public void collect(int doc, long bucket) throws IOException {
+                boolean matched = false;
                 for (int i = 0; i < bits.length; i++) {
                     if (bits[i].get(doc)) {
                         collectBucket(sub, doc, bucketOrd(bucket, i));
+                        matched = true;
                     }
                 }
+                if (!matched) {
+                    collectBucket(sub, doc, bucketOrd(bucket, bits.length));
```

Comment:
Beware that bucketOrd has the following definition:

```java
    final long bucketOrd(long owningBucketOrdinal, int filterOrd) {
        return owningBucketOrdinal * filters.length + filterOrd;
    }
```

So we need to somehow multiply by `filters.length + 1` instead of `filters.length` when we compute the other bucket otherwise there will be several "logical" buckets reusing the same "physical" bucket